### ISSUE TITLE
Kanji shown wrong in the year

### ISSUE THREAD

**Issue by carolinesweetpea** [sopelj/lovelace-kanji-clock-card#18]
**Checklist:**

- [x ] I updated to the latest version available
- [ ] I cleared the cache of my browser

**Release with the issue:**
Latest
**Last working release (if known):**
Unknown
**Browser and Operating System:**
Not browser/OS dependent

**Description of problem:**

The Kanji is wrong :(

2024 is shown as 二千百十四年 (which is actually 2114) 

wherea it should be shown as 二千二十四年

I tried to fix it, however I also can't get it running in a dev environment - `yarn start` doesn't work, and my TS/JS skills aren't good enough at the moment to know how to get things running :( Will keep looking at the issue, and submit a PR if I get htat far.

I think the issue is in line 63 of card.ts - there should be an additional `&& prefix < 10` clause, which is what is putting in the extra 百 but I'm not sure at this stage what is not adding the 二 when it should be in there

**Javascript errors shown in the web inspector (if applicable):**

```

```

**Additional information:**

**Comment by carolinesweetpea** [comment]
This is my test script to fix the output, which I think works (runing with `ts-node test.ts`)

I'll put this into a PR, but it will be untested with HA

```const KANJI_NUMBERS = "霊一二三四五六七八九";
const KANJI_WEEKDAYS = "日月火水木金土";
const TENS = ["", "十", "百", "千"];


const number = 2024

    let output = "";
    const nlength = number.toString().length;
    for (let i = nlength - 1; i >= 0; i--) {
      const divisor = Math.pow(10, i);
      const num = number % divisor;
      const prefix = Math.floor(number / divisor);
      const digit = prefix % 10
      if (digit > 1 && digit < 10) {
	output += KANJI_NUMBERS[digit];
      }
      if (digit > 0) {
        output += TENS[i];
      }
    }
console.log("Output:",output)
```

**Comment by sopelj** [comment]
Good catch! Thanks for that. I'll try testing the PR and merge it this week as soon as I have a moment.

**Comment by sopelj** [comment]
Seems to work well, thanks! I created a beta release if you want to try it.

**Comment by carolinesweetpea** [comment]
Would definitely like to try the beta release - thanks!

**Comment by sopelj** [comment]
@carolinesweetpea did you have a chance to try the beta or version 1.2.0? It seems fine for me.

**Comment by sopelj** [comment]
This should be fixed in 1.2.0, but feel free to open again if still present or create a new one. Thanks.
